sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -24,176 +23,253 @@ OTTF_DEFINE_TEST_CONFIG();
+    OT_ADDRESSABLE_LABEL(kSingleFaultAddrNextPc);
+    addr_val = 2;
   }
-  // I can't add a new variable to call mmio_
-  // because mmio call will never be returned.
-  // Use current variable, just to avoid unused error.
-  addr = mmio_region_read32(mmio_region_from_addr(addr), 0);
+  LOG_ERROR(
+      "This should be unreachable; a reset or another fault should have "
+      "occured.");
 }
 
 /**
- * Overrides the default OTTF exception handler.
+ * Gets, parses and returns the cpu info crash dump.
+ *
+ * @param ibex A handle to the ibex.
+ * @return The cpu info crash dump.
  */
-void ottf_exception_handler(void) {
-  // The exception address ends up being the same since both are
-  // are referencing the same read function
-  temp_dump[kCpuDumpIdxCurrentExceptionPc] =
-      (dif_rstmgr_cpu_info_dump_segment_t)ibex_mepc_read();
-  temp_dump[kCpuDumpIdxCurrentExceptionAddr] = kIllegalAddr2;
-
-  temp_dump[kCpuDumpIdxPreviousExceptionPc] =
-      temp_dump[kCpuDumpIdxCurrentExceptionPc];
-  temp_dump[kCpuDumpIdxPreviousExceptionAddr] = kIllegalAddr1;
-  temp_dump[kCpuDumpIdxPreviousValid] = 1;
-
-  CHECK(flash_ctrl_testutils_write(
-      &flash_ctrl, (uintptr_t)exp_dump - TOP_EARLGREY_FLASH_CTRL_MEM_BASE_ADDR,
-      0, temp_dump, kDifFlashCtrlPartitionTypeData, kCpuDumpSize));
-
-  for (size_t i = 0; i < kCpuDumpSize; ++i) {
-    dif_rstmgr_cpu_info_dump_segment_t rdata = exp_dump[i];
-    LOG_INFO("Expected dump:%d: 0x%x", i, rdata);
-  }
+static dif_rv_core_ibex_crash_dump_info_t get_dump(
+    const dif_rv_core_ibex_t *ibex) {
+  size_t size_read;
+  dif_rstmgr_cpu_info_dump_segment_t dump[DIF_RSTMGR_CPU_INFO_MAX_SIZE];
+
+  CHECK_DIF_OK(dif_rstmgr_cpu_info_dump_read(
+      &rstmgr, dump, DIF_RSTMGR_CPU_INFO_MAX_SIZE, &size_read));
+  CHECK(size_read == kCpuDumpSize,
+        "The observed cpu info dump's size was %d, "
+        "but it was expected to be %d",
+        size_read, kCpuDumpSize);
+
+  dif_rv_core_ibex_crash_dump_info_t output;
+  CHECK_DIF_OK(
+      dif_rv_core_ibex_parse_crash_dump(ibex, dump, size_read, &output));
+  return output;
+}
 
-  read_error();
+/**
+ * Checks the 'current' section of the cpu info dump against the given expected
+ * values.
+ *
+ * @param obs_dump The cpu info crash dump.
+ * @param last_exception_addr The address of the last exception.
+ * @param mpec_lower The expected range lower bound of the last exception PC.
+ * @param mpec_upper The expected range upper bound of the last exception PC.
+ * @param last_data_addr The address of the last data access.
+ * @param curr_pc The current PC.
+ * @param next_pc The last PC.
+ */
+static void check_current_values(dif_rv_core_ibex_crash_dump_info_t *obs_dump,
+                                 uint32_t last_exception_addr,
+                                 uint32_t mpec_lower, uint32_t mpec_upper,
+                                 uint32_t last_data_addr, uint32_t curr_pc,
+                                 uint32_t next_pc) {
```

Comment:
Please use a struct (or define a new one if you cannot use the definitions in `dif_rv_core_ibex`)  instead of defining 6 `uint32_t` parameters to help avoid mistakes. Call-sites then can become:
```
check_current_vaules(foo, (my_struct_t) {
    .last_exception_addr = some_val,
    .mpec_lower = another_val,
    ...
    });
```
Also applies to the function below.

Reply:
Yes, that is far nicer. Changed to this.